fix-engine/src-core-pp/parse_datetime.iml

```diff
@@ -40,7 +40,10 @@ let parse_time_ms str =
         (fun h m s ms ->
           let ms =
             if ms = "" then None
-            else Scanf.sscanf (CCString.pad ~side:`Right ~c:'0' 3i ms) ".%03d" (fun x -> Some (Z.of_int x))
+            else (match CCString.split_on_char '.' ms with
+              | [""; ms] ->
+                Scanf.sscanf (CCString.pad ~side:`Right ~c:'0' 3i ms) "%03d" (fun x -> Some (Z.of_int x))
+              | _ -> failwith "Bad end of string")
```

Comment:
Normally I'd say we don't want to raise an exception here, but `sscanf` will raise anyway so it's not making things worse.

I know `option` doesn't give any error context. The best thing would be to return a `result` but that's too big a refactor for now.